airbyte-webapp/src/components/connectorBuilder/Builder/PaginationSection.tsx

```diff
@@ -0,0 +1,164 @@
+          <ControlLabels
+            label="Pagination"
+            infoTooltipContent="Configure how pagination is handled by your connector"
+          />
+        ),
+        toggledOn,
+        onToggle: handleToggle,
+      }}
+    >
+      <BuilderOneOf
+        path={streamFieldPath("paginator.strategy")}
+        label="Mode"
+        tooltip="Pagination method to use for requests sent to the API"
+        options={[
+          {
+            label: "Offset Increment",
+            typeValue: "OffsetIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Set the size of each page"
+                />
+                {pageSizeOption}
+                {pageTokenOption}
+              </>
+            ),
+          },
+          {
+            label: "Page Increment",
+            typeValue: "PageIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Set the size of each page"
+                />
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.start_from_page")}
+                  label="Start from page"
+                  tooltip="Page number to start requesting pages from"
+                  optional
+                />
+                {pageSizeOption}
+                {pageTokenOption}
+              </>
+            ),
+          },
+          {
+            label: "Cursor Pagination",
+            typeValue: "CursorPagination",
+            children: (
+              <>
+                <BuilderField
+                  type="string"
+                  path={streamFieldPath("paginator.strategy.cursor_value")}
+                  label="Cursor value"
+                  tooltip="Value of the cursor to send in requests to the API"
+                />
+                <BuilderField
+                  type="string"
+                  path={streamFieldPath("paginator.strategy.stop_condition")}
+                  label="Stop condition"
+                  tooltip="Condition that determines when to stop requesting further pages"
+                  optional
+                />
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  onChange={(newValue) => {
+                    if (newValue === undefined || newValue === "") {
+                      pageSizeOptionHelpers.setValue(undefined);
+                    }
+                  }}
+                  label="Page size"
+                  tooltip="Set the size of each page"
+                  optional
+                />
```

Comment:
I decided to keep the page size field here outside of the pageSizeOption group, because it is technically allowed by the low code framework to set a page size here, but not set a page_size_option

Reply:
It's slightly weird that starting to type into the page size input suddenly pops in the page size request options, but I think we can start like this.